src/rendering/colormappingcomponent.cpp

```diff
@@ -257,8 +266,13 @@ ColorMappingComponent::ColorMappingComponent(const ghoul::Dictionary& dictionary
         valueRange = _colorRangeData[dataColumn.value()];
     });
 
-    // @TODO: read valueRange from asset if specified. How to avoid overriding it
-    // in initialize?
+    if (p.parameter.has_value()) {
+        _providedParameter = *(p.parameter);
+    }
+
+    if (p.valueRange.has_value()) {
+        _providedRange = *(p.valueRange);
+    }
```

Comment:
```suggestion
    _providedParameter = p.parameter.value_or(_providedParameter);
    _providedRange = p.valueRange.value_or(_providedRange);
```

?

Reply:
Actually, this does not work.  The `_providedParameter` and `_providedRange` variables are std::optional, and cannot be used as default values for the `value_or` function

Reply:
But setting them to just the value in the parameters work: 
```
    _providedParameter = p.parameter;
    _providedRange = p.valueRange;
```